### ISSUE TITLE
[Issue]: input types 'tensor<1x1x1xf16>' and 'tensor<1xf32>' are not broadcast compatible

### ISSUE THREAD

**Issue by GrinZero** [vladmandic/automatic#824]
### Issue Description

When I execute 'sh webui. sh', the following error message appears.
summary:
I have tried earlier versions of automatic, and at that time M1 Pro could start normally, running 'sh webui. sh' could automatically install and start.
2. When executing this command in the current version, there will be an error (such as the code below).
3. In version [def7a0]( https://github.com/vladmandic/automatic/commit/def7a02d82f4e7456177f3dbdf697dc7d02a3214 ), It can be started successfully, but when clicking to generate an image, the same error will be reported. At this point, setting '--no-half' can execute normally.

```
Create and activate python venv
Launching launch.py...
14:19:20-686199 INFO     Starting SD.Next                                                                                               
14:19:20-688401 INFO     Python 3.10.9 on Darwin                                                                                        
14:19:20-722834 INFO     Version: 4d9fab49 Mon May 8 11:56:59 2023 -0400                                                                
14:19:21-243785 ERROR    Failed to check version: 'commit' {'message': "API rate limit exceeded for 42.98.48.212. (But here's the good  
                         news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",         
                         'documentation_url': 'https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting'}          
14:19:21-244813 INFO     Setting environment tuning                                                                                     
14:19:21-245678 INFO     Using CPU-only Torch                                                                                           
14:19:21-991486 INFO     Torch 2.0.1                                                                                                    
14:19:21-992384 WARNING  Torch repoorts CUDA not available                                                                              
14:19:21-992926 INFO     Installing package: tensorflow==2.12.0                                                                         
14:19:22-498433 INFO     Verifying requirements                                                                                         
14:19:22-521774 INFO     Running setup                                                                                                  
14:19:22-522742 INFO     Installing packages                                                                                            
14:19:22-523189 INFO     Installing package: git+https://github.com/openai/CLIP.git@d50d76daa670286dd6cacf3bcd80b5e4823fc8e1            
14:19:31-149734 INFO     Installing repositories                                                                                        
14:24:08-079444 INFO     Installing submodules                                                                                          
14:27:09-848501 INFO     Updating submodules                                                                                            
14:27:27-758533 INFO     Extensions enabled: ['LDSR', 'Lora', 'ScuNET', 'SwinIR', 'a1111-sd-webui-lycoris', 'clip-interrogator-ext',    
                         'multidiffusion-upscaler-for-automatic1111', 'prompt-bracket-checker', 'sd-dynamic-thresholding',              
                         'sd-extension-aesthetic-scorer', 'sd-extension-steps-animation', 'sd-extension-system-info',                   
                         'sd-webui-controlnet', 'sd-webui-model-converter', 'seed_travel', 'stable-diffusion-webui-images-browser',     
                         'stable-diffusion-webui-rembg']                                                                                
14:28:02-921787 INFO     Extensions enabled: []                                                                                         
14:28:02-923052 INFO     Updating Wiki                                                                                                  
14:28:04-069092 INFO     Running extension preloading                                                                                   
14:28:04-073260 INFO     Server arguments: []                                                                                           
No module 'xformers'. Proceeding without it.
14:28:30-648287 INFO     Libraries loaded                                                                                               
Creating folder: /Volumes/lives/test/stable-diffusion-automatic/models/VAE
Creating folder: /Volumes/lives/test/stable-diffusion-automatic/models/embeddings
Creating folder: outputs/text
Creating folder: outputs/image
Creating folder: outputs/extras
Creating folder: outputs/grids
Creating folder: outputs/save
14:28:30-653337 INFO     Available models: /Volumes/lives/test/stable-diffusion-automatic/models/Stable-diffusion 1                     
ControlNet v1.1.150
ControlNet v1.1.150
Image Browser: ImageReward is not installed, cannot be used.
14:28:46-062984 INFO     No internet access detected, using default fonts                                                               
14:28:46-068316 INFO     Loading theme: black-orange                                                                                    
script_txt2img_tiled_diffusion
script_txt2img_tiled_vae
script_txt2img_dynamic_thresholding_(cfg_scale_fix)
script_txt2img_steps_animation
script_txt2img_controlnet
script_img2img_tiled_diffusion
script_img2img_tiled_vae
script_img2img_dynamic_thresholding_(cfg_scale_fix)
script_img2img_steps_animation
script_img2img_controlnet
Image Browser: Creating database
Image Browser: Database created
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
14:28:47-526573 INFO     Local URL: http://127.0.0.1:7860/                                                                              
14:28:47-527507 INFO     API Docs: http://127.0.0.1:7860/docs                                                                           
14:28:47-527981 INFO     Initializing middleware                                                                                        
14:28:47-572139 WARNING  Default checkpoint not found: model.ckpt                                                                       
14:28:47-572883 WARNING  Loading fallback checkpoint: AnythingV5_v3RE.ckpt                                                              
loc("varianceEps"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/0aa643d0-625a-11ed-b319-a23c4f261b56/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":228:0)): error: input types 'tensor<1x1x1xf16>' and 'tensor<1xf32>' are not broadcast compatible
LLVM ERROR: Failed to infer result type(s).
zsh: abort      sh webui.sh
```

### Version Platform Description

- Version: lastest version([4d9fab4](https://github.com/vladmandic/automatic/commit/4d9fab49848700de2814939228c842b2e0add24d)), Tue, 09 May 2023 06:33:56 GMT
- OS: macos 13.1
- Python: 3.10.9
- GPU: M1Pro

**Comment by marcelreschke** [comment]
Same issue, where can I set the --no-half flag?

**Comment by SoraEnkai** [comment]
Flag settings like `--no-half` are moved to UI, see screenshot:
![Screenshot 2023-05-09 134335](https://github.com/vladmandic/automatic/assets/39128897/6c74a019-13eb-48c5-b22c-bf8e9d83fa76)

**Comment by GrinZero** [comment]
My main confusion is why I have to add - no half, because I didn't actually need it before

**Comment by sullenfish** [comment]
This started happening to me when upgrading from torch 2.0.0 to torch 2.0.1.

My current workaround is to explicitly set torch==2.0.0 and torchvision==0.15.1 in `webui-user.sh`:

`export TORCH_COMMAND="torch==2.0.0 torchvision==0.15.1 --extra-index-url https://download.pytorch.org/whl/rocm5.2"`

Note: I'm running this on an M1 Max with a _highly_ jury-rigged configuration, but that should hopefully get you running again.

**Comment by GrinZero** [comment]
> This started happening to me when upgrading from torch 2.0.0 to torch 2.0.1.
> 
> My current workaround is to explicitly set torch==2.0.0 and torchvision==0.15.1 in `webui-user.sh`:
> 
> `export TORCH_COMMAND="torch==2.0.0 torchvision==0.15.1 --extra-index-url https://download.pytorch.org/whl/rocm5.2"`
> 
> Note: I'm running this on an M1 Max with a _highly_ jury-rigged configuration, but that should hopefully get you running again.

It is indeed the reason for the torch version, but if this code is added in webui.sh, Module clip not found will appear

**Comment by GrinZero** [comment]
@vladmandic 
I seem to have done something wrong. Although I specified the pystore version as' 2.0.0 'in' installer. py ', it seems to only take effect on the first run......During the second run of 'log. info (f'Tarch {torch. __version__}') ', the printed version is' 2.0.1', which still causes errors. This is a scene that I have never encountered before. I would like to ask for your advice.
![image](https://github.com/vladmandic/automatic/assets/70185413/0c5343f1-7cfb-432b-aa5e-6addfc1495a6)

**Comment by vladmandic** [comment]
installer will not force torch version if torch is already present on the system. so if torch got upgraded elsewhere, it will remain as such.

**Comment by GrinZero** [comment]
> installer will not force torch version if torch is already present on the system. so if torch got upgraded elsewhere, it will remain as such.

@vladmandic 
You are right, I checked `setup.log` and found that `torch` has been automatically upgraded after `Installing package: torch==2.0.0 torchaudio torchvision`:
![image](https://github.com/vladmandic/automatic/assets/70185413/606e2a02-63a7-4839-9aac-8911eac154d3)

I find it's because of `install_requirements` function has updated the `torch` function.

![image](https://github.com/vladmandic/automatic/assets/70185413/ae323899-a271-4ab9-8886-85a7b727cc81)

The reason is that although `check_torch` was successfully installed, it still failed to install in `installed ->pkg_resources`. My idea is to remove `torch` and `torch vision` from `requirements.txt` because `check_torch` will install them.
And I have successfully tested them on my computer, removing them from dependencies seems to have no impact.
